apps/strapi-cms/src/api/solution/content-types/solution/schema.json

```
@@ -81,7 +81,8 @@
           "localized": true
         }
       },
-      "component": "common.solution-step"
+      "component": "common.solution-step",
+      "max": 4
```

I think yuo could also add the minimum value of 3 steps:
```suggestion
      "max": 4,
      "min": 3
```